### PR TITLE
RadioGroup Automation Testing

### PR DESCRIPTION
**Pull request by gulnazsayed**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [x] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Adding E2E tests and snapshot testing for RadioGroup and Radio. Added token customization test section to tester. Made the root accessible in RadioGroup so that accessibility works correctly and E2E tests pass.

### Verification

(how the change was tested, including both manual and automated tests)

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | image |

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/experimental/RadioGroup/src/Radio/Radio.styling.ts [hunk 1]
```diff
@@ -52,11 +53,11 @@ export const stylingSettings: UseStylingOptions<RadioProps, RadioSlotProps, Radi
           height: tokens.radioInnerCircleSize,
           width: tokens.radioInnerCircleSize,
           backgroundColor: tokens.radioFill,
-          left: 4,
-          top: 4,
+          left: tokens.radioInnerCircleLeft,
+          top: tokens.radioInnerCircleTop,
```
Comment (Saadnajmi) [inline packages/experimental/RadioGroup/src/Radio/Radio.styling.ts:57]: Any reason we can't use margin/padding? The left/right/top style props always confused me..

**Comment by samuelfreiberg** [comment]
Whenever a PR has two completely separate changes, I'd recommend splitting into two separate PRs: The styling changes and the E2E testing changes.

What happens if we need to revert the styling changes for whatever reason? If it was it's own PR, that would come without consequence (just revert the PR). However, here we'd also need to back out the E2E changes and then re-submit the PR.

It's also important for future devs. Having focused PRs makes it easier to dig through file history and find specific changes, rather than having to look at PRs that contain multiple distinct changes.

**Comment by rurikoaraki** [comment]
> Whenever a PR has two completely separate changes, I'd recommend splitting into two separate PRs: The styling changes and the E2E testing changes.
> 
> What happens if we need to revert the styling changes for whatever reason? If it was it's own PR, that would come without consequence (just revert the PR). However, here we'd also need to back out the E2E changes and then re-submit the PR.
> 
> It's also important for future devs. Having focused PRs makes it easier to dig through file history and find specific changes, rather than having to look at PRs that contain multiple distinct changes.

+1 here. I would prefer to see separate PRs for the changes, one for styling and one for adding tests. The other upside is that if the changes are separated then you don't hold up the entire change if there's an issue with only part of the change.

**Comment by gulnazsayed** [comment]
> Whenever a PR has two completely separate changes, I'd recommend splitting into two separate PRs: The styling changes and the E2E testing changes.
> 
> What happens if we need to revert the styling changes for whatever reason? If it was it's own PR, that would come without consequence (just revert the PR). However, here we'd also need to back out the E2E changes and then re-submit the PR.
> 
> It's also important for future devs. Having focused PRs makes it easier to dig through file history and find specific changes, rather than having to look at PRs that contain multiple distinct changes.

Makes sense! I split up the two PRs. This one is for the testing and the other one is for [styling fixes](https://github.com/microsoft/fluentui-react-native/pull/2301).

File: packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts [hunk 2]
```diff
@@ -35,6 +35,7 @@ export const useRadioGroup = (props: RadioGroupProps): RadioGroupInfo => {
       ...props,
       required,
       layout,
+      accessible: true,
```
Comment (samuelfreiberg) [inline packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts:38]: Does this fix a bug?

I think it's worth mentioning in the PR description :)

Comment (samuelfreiberg) [inline packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts:38]: Does this fix a bug?

I think it's worth mentioning in the PR description :)

File: apps/fluent-tester/src/E2E/RadioGroupExperimental/specs/ExperimentalRadioGroup.spec.win.ts [hunk 3]
```diff
@@ -1 +1,114 @@
...
+
+  it("Keyboard to DISABLED Radio and validate it doesn't get selected", async () => {
+    // Presses the ArrowDown key while the second (B) Radio is selected
+    await RadioGroupExperimentalPageObject.sendKey(Keys.ARROW_DOWN, RadioSelector.Second);
+    await RadioGroupExperimentalPageObject.waitForRadioSelected(RadioSelector.Fourth, 5000); // It should skip RadioButton 3 since it is disabled
+
+    /* Validate the Radio is selected */
+    await expect(await RadioGroupExperimentalPageObject.isRadioSelected(RadioSelector.Fourth)).toBeTruthy();
+    await expect(await RadioGroupExperimentalPageObject.didAssertPopup()).toBeFalsy(RadioGroupExperimentalPageObject.ERRORMESSAGE_ASSERT);
+  });
```
Comment (samuelfreiberg) [inline apps/fluent-tester/src/E2E/RadioGroupExperimental/specs/ExperimentalRadioGroup.spec.win.ts:113]: Let's also add a test case validating circular navigation keyboard behavior

Comment (samuelfreiberg) [reply apps/fluent-tester/src/E2E/RadioGroupExperimental/specs/ExperimentalRadioGroup.spec.win.ts:113]: And that pressing "tab" navigates outside the RadioGroup

**Review by samuelfreiberg — CHANGES_REQUESTED** [review]

**Comment by samuelfreiberg** [comment]
Looks good! Just a couple requests :)

File: packages/experimental/RadioGroup/src/__tests__/RadioGroupExperimental.test.tsx [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
+        <RadioGroup label="Uncontrolled RadioGroup" defaultValue="key2">
+          <Radio value="key1" label="RadioButton1" />
+          <Radio value="key2" label="RadioButton2" />
+        </RadioGroup>,
+      )
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('Radio not direct child of radio group', () => {
```
Comment (rurikoaraki) [inline packages/experimental/RadioGroup/src/__tests__/RadioGroupExperimental.test.tsx:21]: Should there be tests for additional props like disabled?

File: packages/experimental/RadioGroup/src/__tests__/RadioExperimental.test.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { Radio } from '../Radio/Radio';
+import * as renderer from 'react-test-renderer';
+
+describe('Radio component tests', () => {
+  it('Radio default', () => {
+    const tree = renderer.create(<Radio value="key1" label="Default Radio" />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
```
Comment (rurikoaraki) [inline packages/experimental/RadioGroup/src/__tests__/RadioExperimental.test.tsx:9]: We should include rerender tests

File: packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts (hunk 2 above)
Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/RadioGroup/useRadioGroup.ts:38]: Updated, thanks!

**Comment by Saadnajmi** [comment]
If possible, could we extend these tests to macOS? Even though this RadioGroup isn't V1 ready for macOS, the tests would help me personally know that my FocusZone changes don't break one of the components that use it (RadioGroup).

File: apps/fluent-tester/src/E2E/RadioGroupExperimental/specs/ExperimentalRadioGroup.spec.win.ts (hunk 3 above)
Comment (gulnazsayed) [reply apps/fluent-tester/src/E2E/RadioGroupExperimental/specs/ExperimentalRadioGroup.spec.win.ts:113]: Added the test for circular navigation!

File: packages/experimental/RadioGroup/src/__tests__/RadioExperimental.test.tsx (hunk 5 above)
Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/__tests__/RadioExperimental.test.tsx:10]: Added, thanks!

File: packages/experimental/RadioGroup/src/__tests__/RadioGroupExperimental.test.tsx (hunk 4 above)
Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/__tests__/RadioGroupExperimental.test.tsx:21]: Added!

File: apps/fluent-tester/src/E2E/RadioGroupExperimental/specs/ExperimentalRadioGroup.spec.win.ts (hunk 3 above)
Comment (gulnazsayed) [reply apps/fluent-tester/src/E2E/RadioGroupExperimental/specs/ExperimentalRadioGroup.spec.win.ts:113]: Added!

**Comment by gulnazsayed** [comment]
> If possible, could we extend these tests to macOS? Even though this RadioGroup isn't V1 ready for macOS, the tests would help me personally know that my FocusZone changes don't break one of the components that use it (RadioGroup).

@Saadnajmi Absolutely, I just added a few tests for MacOS. I don't have a Mac, so could you run the tests locally to see if they pass? I'd also appreciate it if you could double check that these tests align with expected behavior for MacOS. Thanks!

**Comment by Saadnajmi** [comment]
> > If possible, could we extend these tests to macOS? Even though this RadioGroup isn't V1 ready for macOS, the tests would help me personally know that my FocusZone changes don't break one of the components that use it (RadioGroup).
> 
> @Saadnajmi Absolutely, I just added a few tests for MacOS. I don't have a Mac, so could you run the tests locally to see if they pass? I'd also appreciate it if you could double check that these tests align with expected behavior for MacOS. Thanks!

Read through the functional testing section and it looks good! And macOS E2E tests runs as part of our CI, so ADO should tell us if things broke :)

**Review by rurikoaraki — APPROVED** [review]

**Review by Saadnajmi — APPROVED** [review]

**Review by samuelfreiberg — APPROVED** [review]

File: packages/experimental/RadioGroup/src/__tests__/RadioGroupExperimental.test.tsx [hunk 6]
```diff
@@ -1,21 +1,88 @@
...
+          <Radio value="key1" label="Radio1" accessibilityPositionInSet={1} accessibilitySetSize={2} />
+          <Radio value="key2" label="Radio2" accessibilityPositionInSet={2} accessibilitySetSize={2} />
+        </View>
+      </RadioGroup>,
+    );
+    expect(tree).toMatchSnapshot();
+  });
+
+  // it('RadioGroup disabled', () => {
+  //   const tree = renderer.create(
```
Comment (rurikoaraki) [inline packages/experimental/RadioGroup/src/__tests__/RadioGroupExperimental.test.tsx:34]: Are these tests are expected to work? If there's a build problem then it should be fixed and not commented out.

Comment (gulnazsayed) [reply packages/experimental/RadioGroup/src/__tests__/RadioGroupExperimental.test.tsx:34]: These tests work locally but I'm getting a build error in ADO related to snapshot tests that I'm not getting locally. I'm just commenting out the tests temporarily to pinpoint what I changed that is breaking the build. I'll revert the commenting once I resolve this issue. Thanks!

**Comment by rurikoaraki** [comment]
After syncing offline I don't want this blocked on an issue that seems to be more of a system problem than a problem with the tests themselves, so signing off with some of the unit tests commented out until the issue can be looked into further

**Review by rurikoaraki — APPROVED** [review]

**Comment by gulnazsayed** [comment]
Filed an issue for the commented jest tests: https://github.com/microsoft/fluentui-react-native/issues/2349.

**Pull request merged**